# Non-Linear Least Sqaure Regression

Ordinary Least Square (OLS) is one of the methods to find the best fit line for a dataset using linear regression. The most common application is to create a straight line that minimizes the sum of squares of the errors generated from the differences in the observed value and the vaue anticipated from the model. Least square problems fall into two categories: linear and non-linear squares, depending on whether or not the residuals are linear in all unknowns.

Dataset used:[Kaggle]("https://www.kaggle.com/tmcketterick/job-prestige")

In [1]:
# Importing necessary packages
import pandas as pd
import statsmodels.api as sm

In [2]:
# Reading the dataset
prestige_dataset = pd.read_csv("prestige.csv")
prestige_dataset.head()

job  education  income  women  prestige  census  type
0   gov.administrators      13.11   12351  11.16      68.8    1113  prof
1     general.managers      12.26   25879   4.02      69.1    1130  prof
2          accountants      12.77    9271  15.70      63.4    1171  prof
3  purchasing.officers      11.42    8865   9.11      56.8    1175  prof
4             chemists      14.62    8403  11.68      73.5    2111  prof

In [3]:
# Creating dependent and independent variables from the dataset
x = prestige_dataset.drop(["prestige", "job", "type"], axis = 1)
y = prestige_dataset["prestige"]

In [4]:
x

education  income  women  census
0        13.11   12351  11.16    1113
1        12.26   25879   4.02    1130
2        12.77    9271  15.70    1171
3        11.42    8865   9.11    1175
4        14.62    8403  11.68    2111
..         ...     ...    ...     ...
97        7.58    5562   9.47    9171
98        7.93    4224   3.59    9173
99        8.37    4753   0.00    9313
100      10.00    6462  13.58    9511
101       8.55    3617  70.87    9517

[102 rows x 4 columns]

In [5]:
y

0      68.8
1      69.1
2      63.4
3      56.8
4      73.5
       ... 
97     35.9
98     25.1
99     26.1
100    42.2
101    35.2
Name: prestige, Length: 102, dtype: float64

In [6]:
# Creating Ordinary Least Square (OLS) regression model
ols_model = sm.OLS(y, x).fit()

print(ols_model.summary())

                                 OLS Regression Results                                
Dep. Variable:               prestige   R-squared (uncentered):                   0.976
Model:                            OLS   Adj. R-squared (uncentered):              0.975
Method:                 Least Squares   F-statistic:                              984.6
Date:                Tue, 09 Mar 2021   Prob (F-statistic):                    3.66e-78
Time:                        12:54:36   Log-Likelihood:                         -353.85
No. Observations:                 102   AIC:                                      715.7
Df Residuals:                      98   BIC:                                      726.2
Df Model:                           4                                                  
Covariance Type:            nonrobust                                                  
                 coef    std err          t      P>|t|      [0.025      0.975]
-----------------------------------------

The summary of the model displays the coefficients of the independent variables and value of variance (R-squared) along with the necessary other parameters of the regression model. The value of R-squared is 97.6 which suggest that the total variance explained from the independent variables is 97.6% which means that the model is excellent.